src/main/java/task/Deadline.java

```diff
@@ -0,0 +1,36 @@
+package task;
+
+/**
+ * Deadline task is a specialisation of the Task class that tracks
+ * tasks that need to be done by a specific date and time.
+ */
+public final class Deadline extends Task {
+    /**
+     * Specifies the required argument to create a Deadline.
+     */
+    public static String REQ_ARG = "by";
+
+    /**
+     * When the deadline is.
+     */
+    public String by;
+
+    /**
+     * Creates a new Deadline.
+     *
+     * @param task The task to complete.
+     * @param by   When the task is due by.
+     */
```

Comment:
I agree with the others, the comments make the code clearer and easier to understand.